src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,344 @@
+        , Aria.roleDescription "carousel"
+        , labelAttribute config
+        ]
+    }
+
+
+{-| Builds a carousel with tab buttons
+Returns:
+
+  - `controls`: tabs control buttons
+  - `slides` container with the carousel contents
+
+-}
+viewWithTabControls :
+    { selected : id
+    , slides :
+        List
+            { id : id
+            , idString : String
+            , name : String
+            , visibleLabelId : Maybe String
+            , slideHtml : Html msg
+            , tabControlHtml : Html Never
+            }
+    , tabControlStyles : Bool -> List Style
+    , tabControlListStyles : List Style
+    , role : Role
+    , name : String
+    , visibleLabelId : Maybe String
+    , focusAndSelect : { select : id, focus : Maybe String } -> msg
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithTabControls config =
+    let
+        buildTab tabConfig =
+            TabsInternal.fromList { id = tabConfig.id, idString = tabConfig.idString }
+                [ \tab -> { tab | panelView = tabConfig.slideHtml }
+                , \tab -> { tab | tabView = [ Html.map never tabConfig.tabControlHtml ] }
+                ]
+
+        { tabList, tabPanels } =
+            TabsInternal.views
+                { focusAndSelect = config.focusAndSelect
+                , selected = config.selected
+                , tabs = List.map buildTab config.slides
+                , tabStyles = always config.tabControlStyles
+                , tabListStyles = config.tabControlListStyles
+                }
+    in
+    { controls = tabList
+    , slides = tabPanels
+    , containerAttributes =
+        [ Attrs.attribute "role" (roleToString config.role)
+        , Aria.roleDescription "carousel"
+        , labelAttribute config
+        ]
+    }
+
+
+{-| Builds a carousel with tab buttons
+Returns:
+
+  - `tabControls`: tabs control buttons
+  - `slides` container with the carousel contents
+  - `viewPreviousButton` previous button
+  - `viewNextButton` next button
+
+-}
+viewWithCombinedControls :
+    { selected : id
+    , slides :
+        List
+            { id : id
+            , idString : String
+            , name : String
+            , visibleLabelId : Maybe String
+            , slideHtml : Html msg
+            , tabControlHtml : Html Never
+            }
+    , tabControlStyles : Bool -> List Style
+    , tabControlListStyles : List Style
+    , previousButton : { name : String, icon : Svg, attributes : List (ClickableSvg.Attribute msg) }
+    , nextButton : { name : String, icon : Svg, attributes : List (ClickableSvg.Attribute msg) }
+    , role : Role
+    , name : String
+    , visibleLabelId : Maybe String
+    , focusAndSelect : { select : id, focus : Maybe String } -> msg
+    , announceAndSelect : { select : id, announce : String } -> msg
+    }
+    ->
+        { tabControls : Html msg
```

Comment:
🎨 also super picky, but `viewWithTabs` returns a record with `controls` and this funciton returns a record with `tabControls`. Could we standardize? 🙏